lib/open_project/scm/adapters/git.rb

```diff
@@ -0,0 +1,375 @@
+          parsing_descr = 0 # 0: not parsing desc or files, 1: parsing desc, 2: parsing files
+          parse_by_line(args, binmode: true) do |line|
+            if line =~ /^commit ([0-9a-f]{40})$/
+              key = 'commit'
+              value = $1
+              if parsing_descr == 1 || parsing_descr == 2
+                parsing_descr = 0
+                revision = Revision.new(
+                  identifier: changeset[:commit],
+                  scmid: changeset[:commit],
+                  author: changeset[:author],
+                  time: Time.parse(changeset[:date]),
+                  message: changeset[:description],
+                  paths: files
+                )
+                if block_given?
+                  yield revision
+                else
+                  revisions << revision
+                end
+                changeset = {}
+                files = []
+              end
+              changeset[:commit] = $1
+            elsif (parsing_descr == 0) && line =~ /^(\w+):\s*(.*)$/
+              key = $1
+              value = $2
+              if key == 'Author'
+                changeset[:author] = value
+              elsif key == 'CommitDate'
+                changeset[:date] = value
+              end
+            elsif (parsing_descr == 0) && line.chomp.to_s == ''
+              parsing_descr = 1
+              changeset[:description] = ''
+            elsif (parsing_descr == 1 || parsing_descr == 2) &&
+                  (line =~ /^:\d+\s+\d+\s+[0-9a-f.]+\s+[0-9a-f.]+\s+(\w)\t(.+)$/)
+
+              parsing_descr = 2
+              fileaction = $1
+              filepath = $2
+              p = scm_encode('UTF-8', @path_encoding, filepath)
+              files << { action: fileaction, path: p }
+            elsif (parsing_descr == 1 || parsing_descr == 2) &&
+                  (line =~ /^:\d+\s+\d+\s+[0-9a-f.]+\s+[0-9a-f.]+\s+(\w)\d+\s+(\S+)\t(.+)$/)
+
+              parsing_descr = 2
+              fileaction = $1
+              filepath = $3
+              p = scm_encode('UTF-8', @path_encoding, filepath)
+              files << { action: fileaction, path: p }
+            elsif (parsing_descr == 1) && line.chomp.to_s == ''
+              parsing_descr = 2
+            elsif parsing_descr == 1
+              changeset[:description] << line[4..-1]
+            end
+          end
+
+          if changeset[:commit]
+            revision = Revision.new(
+              identifier: changeset[:commit],
+              scmid: changeset[:commit],
+              author: changeset[:author],
+              time: Time.parse(changeset[:date]),
+              message: changeset[:description],
+              paths: files
+            )
+
+            if block_given?
+              yield revision
+            else
+              revisions << revision
+            end
+          end
+
+          revisions
+        end
+
+        def build_revision_args(path, identifier_from, identifier_to, options)
```

Comment:
Assignment Branch Condition size for build_revision_args is too high. [23.85/15]